### ISSUE TITLE
Error deploying Openshift Origin: [WARNING]: '_es_node' is undefined

### ISSUE THREAD

**Issue by uselessidbr** [openshift/openshift-ansible#12086]
#### Description

Provide a brief description of your issue here. For example:

While deploying Openshift Origin v3.11 the deploy stops on "Logging Install" but the PLAY RECAP shows no failed task.

##### Version

Please put the following version information in the code block
indicated below.

* Your ansible version per `ansible --version`
2.7.5
If you're operating from a **git clone**:

* The output of `git describe`
openshift-ansible-3.11.165-1-12-g958870f
If you're running from playbooks installed via RPM

* The output of `rpm -q openshift-ansible`

Place the output between the code block below:

```
VERSION INFORMATION HERE PLEASE
```

##### Steps To Reproduce
1. Run prereqs playbook
2. Run deploy_cluster playbook

##### Expected Results
Run all playbooks/tasks to the end.

```
****************************************************************************
2020-01-24 18:53:54,649 p=ansible u=24598 | RUNNING HANDLER [openshift_logging_elasticsearch : Checking current health for {{ _es_node }} cluster] **********************************************************************************************
2020-01-24 18:53:54,748 p=ansible u=24598 | [WARNING]: '_es_node' is undefined
2020-01-24 18:53:54,777 p=ansible u=24598 | RUNNING HANDLER [openshift_logging_elasticsearch : Checking current health for {{ _es_node }} cluster] **********************************************************************************************
2020-01-24 18:53:54,783 p=ansible u=24598 | RUNNING HANDLER [openshift_logging_elasticsearch : Checking current health for {{ _es_node }} cluster] **********************************************************************************************

```

##### Observed Results

```
****************************************************************************
2020-01-24 18:53:54,783 p=ansible u=24598 | Friday 24 January 2020  18:53:54 -0300 (0:00:00.000)       0:32:53.342 ********
2020-01-24 18:53:54,783 p=ansible u=24598 | RUNNING HANDLER [openshift_logging_elasticsearch : Checking current health for {{ _es_node }} cluster] ******************
****************************************************************************
2020-01-24 18:53:54,784 p=ansible u=24598 | Friday 24 January 2020  18:53:54 -0300 (0:00:00.000)       0:32:53.342 ********
2020-01-24 18:53:54,784 p=ansible u=24598 | NO MORE HOSTS LEFT ******************************************************************************************************
****************************************************************************
2020-01-24 18:53:54,785 p=ansible u=24598 | PLAY RECAP **************************************************************************************************************
****************************************************************************
2020-01-24 18:53:54,786 p=ansible u=24598 | lnciasc-okd-storage-nfs2.intranet.ciasc.gov.br : ok=24   changed=8    unreachable=0    failed=0    skipped=27   rescued=0
    ignored=0
2020-01-24 18:53:54,786 p=ansible u=24598 | lnciasc-prod-okdlb01.intranet.ciasc.gov.br : ok=27   changed=4    unreachable=0    failed=0    skipped=27   rescued=0
ignored=0
2020-01-24 18:53:54,786 p=ansible u=24598 | lnciasc-prod-okdlb02.intranet.ciasc.gov.br : ok=27   changed=4    unreachable=0    failed=0    skipped=27   rescued=0
ignored=0
2020-01-24 18:53:54,787 p=ansible u=24598 | lnciasc-prod-okdm01.intranet.ciasc.gov.br : ok=1055 changed=421  unreachable=0    failed=0    skipped=967  rescued=0    i
gnored=1
2020-01-24 18:53:54,787 p=ansible u=24598 | lnciasc-prod-okdm02.intranet.ciasc.gov.br : ok=308  changed=169  unreachable=0    failed=0    skipped=412  rescued=0    i
gnored=0
2020-01-24 18:53:54,787 p=ansible u=24598 | lnciasc-prod-okdm03.intranet.ciasc.gov.br : ok=308  changed=166  unreachable=0    failed=0    skipped=412  rescued=0    i
gnored=0
2020-01-24 18:53:54,788 p=ansible u=24598 | lnciasc-prod-okdn01.intranet.ciasc.gov.br : ok=125  changed=75   unreachable=0    failed=0    skipped=153  rescued=0    i
gnored=0
2020-01-24 18:53:54,788 p=ansible u=24598 | lnciasc-prod-okdn02.intranet.ciasc.gov.br : ok=125  changed=75   unreachable=0    failed=0    skipped=153  rescued=0    i
gnored=0
2020-01-24 18:53:54,788 p=ansible u=24598 | lnciasc-prod-okdn03.intranet.ciasc.gov.br : ok=125  changed=75   unreachable=0    failed=0    skipped=153  rescued=0    i
gnored=0
2020-01-24 18:53:54,788 p=ansible u=24598 | lnciasc-prod-okdn04.intranet.ciasc.gov.br : ok=125  changed=75   unreachable=0    failed=0    skipped=153  rescued=0    i
gnored=0
2020-01-24 18:53:54,789 p=ansible u=24598 | localhost                  : ok=13   changed=0    unreachable=0    failed=0    skipped=3    rescued=0    ignored=0
2020-01-24 18:53:54,789 p=ansible u=24598 | INSTALLER STATUS ************************************************************************************************************************************************************************************
2020-01-24 18:53:54,794 p=ansible u=24598 | Initialization               : Complete (0:01:00)
2020-01-24 18:53:54,794 p=ansible u=24598 | Health Check                 : Complete (0:00:48)
2020-01-24 18:53:54,794 p=ansible u=24598 | Node Bootstrap Preparation   : Complete (0:05:10)
2020-01-24 18:53:54,795 p=ansible u=24598 | etcd Install                 : Complete (0:01:03)
2020-01-24 18:53:54,795 p=ansible u=24598 | NFS Install                  : Complete (0:00:12)
2020-01-24 18:53:54,795 p=ansible u=24598 | Load Balancer Install        : Complete (0:00:13)
2020-01-24 18:53:54,796 p=ansible u=24598 | Master Install               : Complete (0:07:05)
2020-01-24 18:53:54,796 p=ansible u=24598 | Master Additional Install    : Complete (0:00:33)
2020-01-24 18:53:54,796 p=ansible u=24598 | Node Join                    : Complete (0:01:29)
2020-01-24 18:53:54,797 p=ansible u=24598 | Hosted Install               : Complete (0:00:57)
2020-01-24 18:53:54,797 p=ansible u=24598 |     The use of NFS for the core OpenShift Container Platform components is not recommended, as NFS (and the NFS Protocol) does not provide the proper consistency needed for the applications that make up the OpenShift Container Platform infrastructure.
2020-01-24 18:53:54,797 p=ansible u=24598 | Cluster Monitoring Operator  : Complete (0:01:14)
2020-01-24 18:53:54,798 p=ansible u=24598 | Web Console Install          : Complete (0:00:43)
2020-01-24 18:53:54,798 p=ansible u=24598 | Console Install              : Complete (0:00:58)
2020-01-24 18:53:54,798 p=ansible u=24598 | Metrics Install              : Complete (0:07:05)
2020-01-24 18:53:54,799 p=ansible u=24598 | metrics-server Install       : Complete (0:00:44)
2020-01-24 18:53:54,799 p=ansible u=24598 | Logging Install              : In Progress (0:03:12)
2020-01-24 18:53:54,799 p=ansible u=24598 |     This phase can be restarted by running: playbooks/openshift-logging/config.yml
2020-01-24 18:53:54,799 p=ansible u=24598 | Friday 24 January 2020  18:53:54 -0300 (0:00:00.015)       0:32:53.358 ********
2020-01-24 18:53:54,800 p=ansible u=24598 | ===============================================================================

```

For long output or logs, consider using a [gist](https://gist.github.com/)

##### Additional Information

Provide any additional information which may help us diagnose the
issue.

* Your operating system and version, ie: RHEL 7.2, Fedora 23 (`$ cat /etc/redhat-release`)
* Your inventory file (especially any non-standard configuration parameters)
* Sample code, etc

```
CentOS Linux release 7.7.1908 (Core)

[OSEv3:children]
masters
nodes
etcd
lb
nfs

## Set variables common for all OSEv3 hosts
[OSEv3:vars]
ansible_ssh_user=ansible
ansible_become=true
openshift_deployment_type=origin
openshift_release=v3.11

openshift_master_cluster_method=native

openshift_portal_net=10.1.128.0/18
osm_cluster_network_cidr=10.1.0.0/17
osm_host_subnet_length=9

openshift_use_calico=True
openshift_use_openshift_sdn=False
os_sdn_network_plugin_name='cni'

openshift_console_install=true
openshift_console_hostname=console.openshift.local

openshift_master_cluster_hostname=int.openshift.local

openshift_master_cluster_public_hostname=okd.openshift.local

openshift_master_default_subdomain=apps.okd.openshift.local

openshift_use_crio=True
openshift_use_crio_only=False
openshift_crio_enable_docker_gc=True

openshift_docker_options=--bip 10.0.0.1/24 --log-opt  max-size=100M --log-opt max-file=3 --insecure-registry 10.1.128.0/17 --insecure-registry 10.0.0.0/24 --log-driver=json-file --authorization-plugin=docker-novolume-plugin


## Autenticacao
openshift_master_identity_providers=[{'name': 'Local Authentication', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider'}]

openshift_master_htpasswd_users={'admin': '$apr1$4qJTE7h9$j9zVzh43pFMjaCa/wuVlY.', 'developer': '$apr1$VtHG.FnT$b0XJ3355yxtDzqtiwb7Ag/' }

os_firewall_use_firewalld=true

openshift_hosted_registry_cert_expire_days=3650
openshift_ca_cert_expire_days=5475
openshift_node_cert_expire_days=3650
openshift_master_cert_expire_days=3650
etcd_ca_default_days=5475

openshift_master_dynamic_provisioning_enabled=true

openshift_enable_unsupported_configurations=True

openshift_hosted_registry_storage_kind=nfs
openshift_hosted_registry_storage_access_modes=['ReadWriteMany']
openshift_hosted_registry_storage_host=nfs.openshift.local
openshift_hosted_registry_storage_nfs_directory=/exports
openshift_hosted_registry_storage_nfs_options='*(rw,root_squash)'
openshift_hosted_registry_storage_volume_name=registry
openshift_hosted_registry_storage_volume_size=10Gi

openshift_metrics_install_metrics=true
openshift_metrics_hawkular_nodeselector={"node-role.kubernetes.io/infra": "true"}
openshift_metrics_cassandra_nodeselector={"node-role.kubernetes.io/infra": "true"}
openshift_metrics_heapster_nodeselector={"node-role.kubernetes.io/infra": "true"}
openshift_metrics_storage_kind=nfs
openshift_metrics_storage_access_modes=['ReadWriteOnce']
openshift_metrics_storage_host=nfs.openshift.local
openshift_metrics_storage_nfs_directory=/exports
openshift_metrics_storage_volume_name=metrics
openshift_metrics_storage_volume_size=10Gi

openshift_logging_install_logging=true
openshift_logging_storage_kind=nfs
openshift_logging_kibana_nodeselector={"node-role.kubernetes.io/infra": "true"}
openshift_logging_curator_nodeselector={"node-role.kubernetes.io/infra": "true"}
openshift_logging_es_nodeselector={"node-role.kubernetes.io/infra": "true"}
openshift_logging_storage_access_modes=['ReadWriteOnce']
openshift_logging_storage_nfs_options='*(rw,root_squash)'
openshift_logging_storage_host=nfs.openshift.local
openshift_logging_storage_nfs_directory=/exports
openshift_logging_storage_volume_name=logging
openshift_logging_storage_volume_size=10Gi
openshift_logging_elasticsearch_storage_type=pvc
openshift_logging_es_pvc_size=10Gi
openshift_logging_es_pvc_storage_class_name=''
openshift_logging_es_pvc_dynamic=true
openshift_logging_es_pvc_prefix=logging
openshift_logging_es_memory_limit=4Gi

openshift_node_groups=[{'name': 'node-config-master-crio', 'labels': ['node-role.kubernetes.io/master=true', 'runtime=cri-o'], 'edits': [{ 'key': 'kubeletArguments.container-runtime','value': ['remote'] }, { 'key': 'kubeletArguments.container-runtime-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.image-service-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.runtime-request-timeout','value': ['10m'] }]},{'name': 'node-config-infra-crio', 'labels': ['node-role.kubernetes.io/infra=true', 'runtime=cri-o'], 'edits': [{ 'key': 'kubeletArguments.container-runtime','value': ['remote'] }, { 'key': 'kubeletArguments.container-runtime-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.image-service-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.runtime-request-timeout','value': ['10m'] }]},{'name': 'node-config-compute-crio', 'labels': ['node-role.kubernetes.io/compute=true', 'runtime=cri-o'], 'edits': [{ 'key': 'kubeletArguments.container-runtime','value': ['remote'] }, { 'key': 'kubeletArguments.container-runtime-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.image-service-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.runtime-request-timeout','value': ['10m'] }]},{'name': 'node-config-master-infra-crio', 'labels': ['node-role.kubernetes.io/infra=true', 'node-role.kubernetes.io/master=true', 'node-role.kubernetes.io/compute=true', 'runtime=cri-o'], 'edits': [{ 'key': 'kubeletArguments.container-runtime','value': ['remote'] }, { 'key': 'kubeletArguments.container-runtime-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.image-service-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.runtime-request-timeout','value': ['10m'] }]},{'name': 'node-config-all-in-one-crio', 'labels': ['node-role.kubernetes.io/infra=true', 'node-role.kubernetes.io/compute=true' ,'node-role.kubernetes.io/master=true', 'node-role.kubernetes.io/compute=true', 'runtime=cri-o'], 'edits': [{ 'key': 'kubeletArguments.container-runtime','value': ['remote'] }, { 'key': 'kubeletArguments.container-runtime-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.image-service-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.runtime-request-timeout','value': ['10m'] }]},{'name': 'node-config-all-in-one', 'labels': ['node-role.kubernetes.io/master=true', 'node-role.kubernetes.io/infra=true', 'node-role.kubernetes.io/compute=true']}, {'name': 'node-config-master-infra', 'labels': ['node-role.kubernetes.io/master=true', 'node-role.kubernetes.io/infra=true']}, {'name': 'node-config-compute', 'labels': ['node-role.kubernetes.io/compute=true']}, {'name': 'node-config-master', 'labels': ['node-role.kubernetes.io/master=true']}, {'name': 'node-config-infra', 'labels': ['node-role.kubernetes.io/infra=true']}, {'name': 'node-config-compute-crio-prod', 'labels': ['node-role.kubernetes.io/compute=true', 'runtime=cri-o', 'node-role.kubernetes.io/environment=prod'], 'edits': [{ 'key': 'kubeletArguments.container-runtime','value': ['remote'] }, { 'key': 'kubeletArguments.container-runtime-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.image-service-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.runtime-request-timeout','value': ['10m'] }]}, {'name': 'node-config-compute-crio-stage', 'labels': ['node-role.kubernetes.io/compute=true', 'runtime=cri-o', 'node-role.kubernetes.io/environment=stage', 'node-role.kubernetes.io/build=true'], 'edits': [{ 'key': 'kubeletArguments.container-runtime','value': ['remote'] }, { 'key': 'kubeletArguments.container-runtime-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.image-service-endpoint','value': ['/var/run/crio/crio.sock'] }, { 'key': 'kubeletArguments.runtime-request-timeout','value': ['10m'] }]}]

osm_default_node_selector='node-role.kubernetes.io/environment=stage'
openshift_builddefaults_nodeselectors={'node-role.kubernetes.io/build': 'true'}

[nfs]
nfs.openshift.local

# host group for etcd
[etcd]
m01.openshift.local
m02.openshift.local
m03.openshift.local

[masters]
m01.openshift.local
m02.openshift.local
m03.openshift.local

[lb]
lb01.openshift.local

[nodes]
m0[1:3].openshift.local openshift_node_group_name='node-config-master-infra-crio'
n0[1:2].openshift.local openshift_node_group_name='node-config-compute-crio-prod'
n0[3:4].openshift.local openshift_node_group_name='node-config-compute-crio-stage'
```

**Comment by uselessidbr** [comment]
Im guessing it’s something related to the ansible server and it’s modules since i get an error while trying to install the requirements (pip install -r requirements.txt). Also, there’s some python warnings in the log.

**Comment by uselessidbr** [comment]
#12016 (requirements.txt) seems to be the problem here.

Using Ansible 2.8.1 or 2.6 doesn't give any error.

Altough, for some reason, the fluentd is in CrashLoopBack state as it cant't find /etc/fluent/metrics/tls.crt

2020-01-25 16:24:03 -0300 [error]: unexpected error error_class=Errno::ENOENT error="No such file or directory @ rb_sysopen - /etc/fluent/metrics/tls.crt"
  2020-01-25 16:24:03 -0300 [error]: suppressed same stacktrace
2020-01-25 16:24:03 -0300 [error]: fluent/log.rb:362:error: unexpected error error_class=Errno::ENOENT error="No such file or directory @ rb_sysopen - /etc/fluent/metrics/tls.crt"
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluent-plugin-prometheus-1.3.0/lib/fluent/plugin/in_prometheus.rb:68:in `read'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluent-plugin-prometheus-1.3.0/lib/fluent/plugin/in_prometheus.rb:68:in `start'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/root_agent.rb:203:in `block in start'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/root_agent.rb:192:in `block (2 levels) in lifecycle'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/root_agent.rb:191:in `each'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/root_agent.rb:191:in `block in lifecycle'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/root_agent.rb:178:in `each'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/root_agent.rb:178:in `lifecycle'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/root_agent.rb:202:in `start'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/engine.rb:274:in `start'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/engine.rb:219:in `run'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/supervisor.rb:805:in `run_engine'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/supervisor.rb:549:in `block in run_worker'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/supervisor.rb:730:in `main_process'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/supervisor.rb:544:in `run_worker'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/lib/fluent/command/fluentd.rb:316:in `<top (required)>'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:59:in `require'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:59:in `require'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.4.1/bin/fluentd:8:in `<top (required)>'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/bin/fluentd:23:in `load'
  2020-01-25 16:24:03 -0300 [error]: fluent/supervisor.rb:729:main_process: /opt/rh/rh-ruby25/root/usr/local/bin/fluentd:23:in `<main>'

**Comment by uselessidbr** [comment]
Had to uninstall ansible installed by yum and then download pip. After that, I run pip install -r requirements.txt and the error is gone.

But I found that the less error prone version for ansible (installed via pip) is 2.6 because on any other version 2.8.x or 2.9.x it doesn’t create the PV for registry when using registry with NFS.

**Comment by mtnbikenc** [comment]
Fixed in #12094.
